system/lib/wasmfs/syscalls.cpp

```diff
@@ -861,6 +861,58 @@ int __syscall_rmdir(intptr_t path) {
   return __syscall_unlinkat(AT_FDCWD, path, AT_REMOVEDIR);
 }
 
+// wasmfs_unmount is similar to __syscall_unlinkat, but assumes AT_REMOVEDIR is true and will unlink nonempty directories.
+int wasmfs_unmount(int dirfd, intptr_t path) {
+  std::string_view p((char*)path);
+  // Ignore trailing '/'.
+  while (!p.empty() && p.back() == '/') {
+    p.remove_suffix(1);
+  }
+  if (p.size() >= 2 && p.substr(p.size() - 2) == std::string_view("/.")) {
+    return -EINVAL;
+  }
```

Comment:
I'm surprised that you would have to do any custom path parsing here. What is this for, and can we remove it?

Reply:
I copied this directly from `__syscall_unlinkat`, since `wasmfs_unmount` is pretty much the same functionality with the exception of checking that the directory is a mountpoint and allowing nonempty unlinks. To be honest, I'm not entirely sure what it does. Do you think it would be worth keeping here?

Reply:
Ah, I see. `rmdir`'s man page specifically mentions returning errors when paths end in `/.`, which is what this code checks for. `umount`'s man page does not mention these errors and of course we have some flexibility to do what we want in our JS API, so I think we can go ahead and delete this code. If this does end up being important for some reason, then it would be good to factor it out into a shared utility in paths.h.